docs/design/reserve-action.md

```diff
@@ -0,0 +1,273 @@
+# Reserve Action
+
+[@sivanzcw](https://github.com/sivanzcw); July 13, 2019
+
+## Table of Contents
+
+- [Reserve Action](#reserve-action)
+  * [Table of Contents](#table-of-contents)
+  * [Introduction](#introduction)
+  * [Motivation](#motivation)
+  * [Function Detail](#function-detail)
+    + [Terminology](#terminology)
+      - [Job Ready](#job-ready)
+      - [Job Condition Ready](#job-condition-ready)
+      - [Job starving](#job-starving)
+      - [ForceAllocated](#forceallocated)
+      - [FakeAllocated](#fakeallocated)
+    + [Node resource reservation rule](#node-resource-reservation-rule)
+      - [Node reserve logic](#node-reserve-logic)
+      - [The proportion of nodes participating in reserve action](#the-proportion-of-nodes-participating-in-reserve-action)
+      - [Node selection strategy to participate in reserve](#node-selection-strategy-to-participate-in-reserve)
+      - [Node calculating score logic](#node-calculating-score-logic)
+    + [Reserve logic](#reserve-logic)
+  * [Data Model](#data-model)
+
+## Introduction
+
+The `reserve` action is to solve the problem that the task can not be scheduled for a long time. We call these long 
+unscheduled jobs starving jobs, if a job is not scheduled for a long time, it is a starving job. The `reserve` action 
```

Comment:
s/,/; for `unscheduled jobs starving jobs, if a job is not sc`

Reply:
updated it